lta/picker.py

```diff
@@ -79,24 +86,116 @@ def __init__(self, config: Dict[str, str], logger: Logger) -> None:
+            self.logger.info(f"No TransferRequests are available to work on. Going on vacation.")
+            return
+        self.logger.info(f"There are {len(results)} TransferRequest(s) to work on.")
+        # for each TransferRequest that we were given
+        for tr in results:
+            await self._do_work_transfer_request(lta_rc, tr)
+        # log a friendly message
+        self.logger.info(f'Done working on all TransferRequests.')
+
+    async def _do_work_transfer_request(self, lta_rc, tr) -> None:
+        self.logger.info(f"Processing TransferRequest: {tr}")
+        # configure a RestClient to talk to the File Catalog
+        fc_rc = RestClient(self.file_catalog_rest_url,
+                           token=self.file_catalog_rest_token,
+                           timeout=self.work_timeout_seconds,
+                           retries=self.work_retries)
+        # figure out which files need to go
+        src_split = tr['source'].split(':', 1)
+        src_site = src_split[0]
+        src_path = src_split[1]
+        # figure out where they need to go to
+        dests = []
+        for dst in tr['dest']:
+            dst_split = dst.split(':', 1)
+            dests.append((dst_split[0], dst_split[1]))
+        # query the file catalog for the source files
+        self.logger.info(f"Asking the File Catalog about files in {src_split}:{src_path}")
+        query_dict = {
+            "locations.site": {
+                "$eq": f"{src_site}"
+            },
+            "locations.path": {
+                "$regex": f"^{src_path}"
+            }
+        }
+        query_json = json.dumps(query_dict)
+        fc_response = await fc_rc.request('GET', f'/api/files?query={query_json}')
+        self.logger.info(f'File Catalog returned {len(fc_response["files"])} file(s) to process.')
+        # for each file provided by the catalog
+        bulk_create = []
+        for catalog_file in fc_response["files"]:
+            await self._do_work_catalog_file(lta_rc, tr, fc_rc, dests, catalog_file, bulk_create)
+        # 3. Update the REST DB with Files needed for bundling
+        self.logger.info(f'Identified {len(bulk_create)} transfer(s) to add to the REST DB.')
+        create_body = {
+            "files": bulk_create
+        }
+        await lta_rc.request('POST', '/Files/actions/bulk_create', create_body)
+        # 4. Return the TransferRequest to the REST DB as picked
+        self.logger.info(f'Marking TransferRequest {tr["uuid"]} as complete in the REST DB.')
+        complete = {
+            "complete": {
+                "timestamp": datetime.utcnow().isoformat(),
+                "picker": self.picker_name
+            }
+        }
+        await lta_rc.request('PATCH', f'/TransferRequests/{tr["uuid"]}', complete)
+        self.logger.info(f'Done working on TransferRequest {tr["uuid"]}.')
+
+    async def _do_work_catalog_file(self, lta_rc, tr, fc_rc, dests, catalog_file, bulk_create) -> None:
```

Comment:
It's bad form to have mutating arguments, such as `bulk_create`.  Better to return a list.